tool/src/atsp.dart

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// See: Asymmetric Traveling Salesman Problem.
+
+// Strategy for finding optimal overlapping of chunks of a larger table,
+// to save space.
+// Does so by solving traveling salesman/hamiltonian cycle in a graph
+// where the distance between chunks is how little they overlap
+// (chunk length minus overlap size).
+
+/// An asymmetric weighted complete graph.
+///
+/// The vertices are identified by numbers 0 through [vertices] - 1.
+/// Edges are pairs of vertices.
+class Graph {
+  /// Number of vertices.
+  final int vertices;
```

Comment:
[optional] How about `vertexCount`?

Reply:
The way the class is written, the integers *are* the vertices.

But done.